app/views/dashboard/_node_meta.html.erb

```diff
@@ -20,7 +20,10 @@
     <span class="hidden-xs hidden-sm">| <i class="fa fa-eye"></i> <%= number_with_delimiter(node.totalviews) %></span>
     | <i style="<% if node.likes > 0 %>color:#db4;<% end %>" class="fa fa-star-o"></i> <%= node.likes %>
   <% if params[:mod] %>| <a href="#"><i class="fa fa-ban"></i></a><% end %>
-  <a rel="tooltip" title="Flag as spam." class="btn btn-xs btn-default btn-flag-spam-<%= node.id %>" href="mailto:moderators@publiclab.org?subject=Reporting+spam+on+Public+Lab&body=Hi,+I+found+this+item+that+looks+like+spam+or+needs+to+be+moderated:+<%= node.title.gsub(/ /,'+') %>+https://publiclab.org/n/<%= node.id %>+by+https://publiclab.org/profile/<%= node.author.username %>+Thanks!">
-    <i class="fa fa-flag"></i>
-  </a>
+  <% if current_user %>
+    <a class="btn btn-default btn-xs" rel="popover" data-container="body" data-placement="top" data-template='<div class="popover" role="tooltip"><div class="arrow"></div><div class="popover-body"><%= render partial:'dashboard/flag_list', locals: { node: node } %></div></div>' data-html="true" data-content="sample" > <i class="fa fa-flag"> </i> </a>
```

Comment:
Oh my gosh! I get it! The issue is that Bootstrap pop-overs are generated with JavaScript, and the tests don't run javascript (we have separate JS tests for that!) So although the code is here, it's stuffed inside the `data-template` attribute!

We could change the test to something like:

```ruby
    assert_select 'a[data-template=?]', "popover"
```

and be OK with not checking the exact text, OR we could try scanning the entire page for the string we want, rather than looking just for the CSS selector. Does this make sense?

Eureka! 🎉

Reply:
assert_select 'a[data-template=?]', "popover" I think this should be right solution to it, the rest of the code looks fine to me, also, finally which link I should user the mail thing? Thank you so so much, amazing debugging there!

Reply:
I think this one:
"mailto:moderators@publiclab.org?subject=Reporting+spam+on+Public+Lab&body=Hi,+I+found+this+item+that+looks+like+spam+or+needs+to+be+moderated:+<%= node.title.gsub(/ /,'+') %>+https://publiclab.org/n/<%= node.id %>+by+https://publiclab.org/profile/<%= node.author.username %>+Thanks!"

looks fine, because this is what originally the "Flag" button was linked to. What do you think?